src/Fractions/Comparers/FractionValueEqualityComparer.cs

```diff
@@ -22,19 +35,71 @@ public sealed class FractionValueEqualityComparer : FractionComparer {
             return false;
         }
 
-        if (numerator1.IsZero && numerator2.IsZero) {
-            // both values are 0
-            return true;
+        if (numerator1.IsZero) {
+            // both values should be 0 
+            return numerator2.IsZero;
+        }
+
+        if (numerator2.IsZero) {
+            // both values should be 0 
+            return false;
+        }
+
+        var firstNumeratorSign = numerator1.Sign;
+        var secondNumeratorSign = numerator2.Sign;
+        if (firstNumeratorSign != secondNumeratorSign) {
+            return false; // different signs
+        }
+
+        // both values are non-zero fractions with different denominators
+        if (denominator1 < denominator2) {
+            // reverse the comparison from x == y to y == x
+            (numerator1, numerator2) = (numerator2, numerator1);
+            (denominator1, denominator2) = (denominator2, denominator1);
+        }
+
+        // [denominator1 > denominator2 > 0]
+        if (firstNumeratorSign == 1) {
+            // example: {10/10} and {1/1} or {10/100} and {1/10}
+            if (numerator1 <= numerator2) {
+                return false; // expecting: 0 < numerator2 < numerator1
+            }
+
+            if (numerator2.IsOne) {
+                return denominator2.IsOne ? numerator1 == denominator1 : numerator1 * denominator2 == denominator1;
+            }
+        } else {
+            // example: {-10/10} and {-1/1} or {-10/100} and {-1/10}
+            if (numerator1 >= numerator2) {
+                return false; // expecting: numerator1 < numerator2 < 0
+            }
+
+            if (numerator2 == BigInteger.MinusOne) {
+                return denominator2.IsOne ? numerator1 == -denominator1 : numerator1 * -denominator2 == denominator1;
+            }
         }
 
-        if (numerator1.IsZero || numerator2.IsZero) {
-            // either x or y has the value 0
+        if (denominator2.IsOne) {
+            return numerator1 == numerator2 * denominator1;
+        }
+
+        var numeratorQuotient = BigInteger.DivRem(numerator1, numerator2, out var remainderNumerators);
+        var denominatorQuotient = BigInteger.DivRem(denominator1, denominator2, out var remainderDenominators);
+        // if the fractions are equal: numeratorQuotient should be equal to denominatorQuotient
+        if (numeratorQuotient != denominatorQuotient) {
```

Comment:
This is the most interesting part:
What we're doing is taking a roundabout way of comparing the fractions by checking if their `ratio` is exactly 1 : 
`A / B == 1` -> `A == B`
Expressing this using the 4 terms gives us `{n1, d1} / {n2, d2} = {(n1 * d2)  / (d1 * n2)} = {n1 / n2} * {d2 / d1} `
When we set `{n1 / n2} * {d2 / d1} = 1`  (and having previously ensured that the terms are non-negative)
It follows that `{d2 / d1} = {n2 / n1}` should be true (in order for A to be equal to B)

From here on we do the standard DivRem comparison (split into separate operations here, for simplicity):
`{d2 / d1} = {n2 / n1}` => 
1. BigInteger.Divide(d2, d1) == BigInteger.Divide(n2, n1)  // the quotients should be equal
2. `(d2 % d1) / d1 == (n2 % n1) / n1`  // the remainders (r1, r2) expressed as a ratio (Fraction) of their respective dividend should be equal

If all conditions along the way are satisfied, we still end up comparing the product of 2 terms, but for most other cases we exit early. 
Furthermore, the reason for selecting this particular set of operations is not accidental (it would have been _simpler_ to just compare `{n1/d1}` and `{n2/d2}`): the reason is that if the two fractions are actually equal, then the result of `dividing` the two fractions would be something like {10/10} or {200/200} (_but never {1/1} since d1>d2_). In any way, this value is almost certainly _not huge_ and the final 2 products (`remainderNumerators * denominator2`  and `remainderDenominators * numerator2`) would have values that are smaller that the original cross-term multiplication (due to `remainderDenominators < denominator2 < denominator1`).

Reply:
> {(n1 * d2) / (d1 * n2)} = {n1 / n2} * {d2 / d1}

Ah, after reproducing the calculation on paper (or unforming the formula accordingly), I see that this equation is correct. Thanks.

Reply:
FYI - the code reviews take longer because I can't always understand your thought process based on the code (especially when mathematical equations need to be transformed).